pkg/cluster/config/types.go

```diff
@@ -75,3 +108,52 @@ type LifecycleHook struct {
 	// the boot process will continue
 	MustSucceed bool
 }
+
+// IsControlPlane returns true if the node hosts a control plane instance
+func (n *Node) IsControlPlane() bool {
```

Comment:
this is something that i've mentioned earlier - are public methods for root internal config objects something we want?